### ISSUE TITLE
Add "change" keyword in parallel managed lanes

### ISSUE THREAD

**Issue by jhelsel11** [wsp-sag/network_wrangler#162]
It would be super helpful to be able to use the change keyword in setting the lane property in a managed lane project. That way the user wouldn't need to know the number of lanes/create a link id group for each lane width over the course of a project. 

I am experimenting with putting both the managed lane add and roadway property change in a single project card and if that works I will try to look at adding the change switch, but wanted you to be aware.

**Comment by e-lo** [comment]
Reassigning this to @kulshresthaa 

(At least) three code chunks that need to be fixed:

https://github.com/wsp-sag/network_wrangler/blob/4fae96b0ee9785639587ceae2fe196c727097e3d/network_wrangler/RoadwayNetwork.py#L1238-L1240

https://github.com/wsp-sag/network_wrangler/blob/4fae96b0ee9785639587ceae2fe196c727097e3d/network_wrangler/RoadwayNetwork.py#L1253-L1255

https://github.com/wsp-sag/network_wrangler/blob/4fae96b0ee9785639587ceae2fe196c727097e3d/network_wrangler/RoadwayNetwork.py#L1262-L1262

In addition to allowing for change, please update it so that it doesn't overwrite any existing values for categories/time periods that might exist!

**Comment by kulshresthaa** [comment]
@jhelsel11 You have an example project card with "change" keyword for the managed lane project? How is "change" going to be different than "set"?

**Comment by jhelsel11** [comment]
Change should override the current value wityan increment rather than set the value. That way if a road changes lanes from 4 to 3 to 4 and you want one of them to be an hov, you don't need to know which lanes started with which value

**Comment by kulshresthaa** [comment]
I see. So you want "change" as keyword in manage lane card, the same way we have it defined in other cards functionality (roadway attribute change).

**Comment by jhelsel11** [comment]
Yes

**Comment by kulshresthaa** [comment]
@e-lo Not sure if this is straightforward. 

Example project card for discussion:
  - property: lanes
    set: 3
    timeofday:
      - time: ['6:00', '9:00']
        set: 2
  - property: ML_lanes
    set: 0
    timeofday:
      - time: ['6:00', '9:00']
        set: 1
  - property: ML_access
    set: 'all'
  - property: ML_egress
    set: 'all'

If attribute doesn't exist in the network:

- "change" is specified in the project card, throw an error

if attribute exist in the network: 
- There is no "timeofday" or "group" keyword for the attribute: 
simple. just read the "change" value and modify the attribute value as existing + change
- There is "timeofday" keyword:
        Say for "lanes", the attribute value that gets formed is something like `{'default': 3, 'timeofday': [{'time': (21600, 32400), 'value': 2}]}`
        How would you do it with change?


In the current implementation, I don't see this as simple change. We only specify the default value and value by time of day (and by group) and the actual managed lane network isn't created.

Not sure how/where revised value (existing + change) would be computed for each of the specified (search) link. Need to discuss this one.

**Comment by e-lo** [comment]
For now this is "won't fix" for this round. Sounds like Met Council doesn't have any variation in properties across time of day, etc. other than managed lanes.

**Comment by jhelsel11** [comment]
@e-lo, did we get a new name for HOV?

**Comment by e-lo** [comment]
Segment_id

**Comment by kulshresthaa** [comment]
@DavidOry @e-lo This is the issue for adding the 'change' keyword in the parallel managed lane, which we decided to not fix at the time because it is not straightforward.

In the current implementation, the way the new attribute value is constructed (for the time of day), I don't see how to make the change. Perhaps we need to have a call to discuss how to approach this.

I am thinking we likely have to loop through the links and make the change one link at a time by constructing that link attribute value based on set or change.

**Comment by e-lo** [comment]
@kulshresthaa I am juggling things so can't have a call today but can take a look this evening. Can you reference where in the code this would need to happen?

**Comment by kulshresthaa** [comment]
[this](https://github.com/wsp-sag/network_wrangler/blob/master/network_wrangler/roadwaynetwork.py#L1345) is code block where the change needs to happen and [this](https://github.com/wsp-sag/network_wrangler/blob/master/network_wrangler/roadwaynetwork.py#L1399) is where the value is currently set.

current code block - 

* for each property in project card:
  * construct the same attribute value using set and time-of-day keyword
  * loop selected link idx:
    * set the attribute value for the link

how I think it should be revised to account for change keyword - 

* for each property in project card:
  * loop selected link idx:
    * construct the link specific attribute value using set/change and time-of-day keyword
    * set the attribute value for the link

**Comment by kulshresthaa** [comment]
@e-lo  thoughts?

**Comment by kulshresthaa** [comment]
@e-lo please review when you get a chance.

**Comment by DavidOry** [comment]
@kulshresthaa 

When I apply a Parallel Managed Lane Project Card, the `lanes` attribute becomes a dictionary, segmented by time of day. If I then apply another `change` project card to this facility, the `change` integer is added to the `lanes` dictionary, which throws a bug. Example is [here](https://github.com/wsp-sag/network_wrangler/blob/develop/notebook/change-functionality-bug.ipynb). 

cc: @i-am-sijia

**Comment by e-lo** [comment]
moot - as parallel managed lanes is now a roadway property change in v1 branch: https://github.com/wsp-sag/network_wrangler/pull/327
